library/alloc/src/string.rs

```diff
@@ -1593,6 +1593,13 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    #[unstable(feature = "liballoc_internals", issue = "none", reason = "implementation detail")]
+    #[doc(hidden)]
+    pub fn __push_fmt(&mut self, args: fmt::Arguments<'_>) {
+        // Writing to a `String` can never fail.
+        let _ = fmt::Write::write_fmt(self, args);
```

Comment:
The formatting itself might fail and the format macro panics in that case. Is different error handling strategy here intentional?

Reply:
Good catch, yeah, this should unwrap instead.